internal/pkg/cli/storage_init.go

```diff
@@ -692,65 +691,63 @@ func (o *initStorageOpts) readWorkloadType() error {
 	blob        encoding.BinaryMarshaler
+	path        string
 }
 
 func (o *initStorageOpts) addonBlobs() ([]addonBlob, error) {
-	templateBlob, err := o.newAddonTemplate()
+	switch {
+	case o.lifecycle == lifecycleWorkloadLevel && o.storageType == s3StorageType:
+		return o.wkldS3AddonBlobs()
+	case o.lifecycle == lifecycleWorkloadLevel && o.storageType == dynamoDBStorageType:
+		return o.wkldDDBAddonBlobs()
+	case o.lifecycle == lifecycleWorkloadLevel && o.storageType == rdsStorageType:
+		return o.wkldRDSAddonBlobs()
+	case o.lifecycle == lifecycleEnvironmentLevel && o.storageType == s3StorageType:
+		return o.envS3AddonBlobs()
+	case o.lifecycle == lifecycleEnvironmentLevel && o.storageType == dynamoDBStorageType:
+		return o.envDDBAddonBlobs()
+	case o.lifecycle == lifecycleEnvironmentLevel && o.storageType == rdsStorageType:
+		return o.envRDSAddonBlobs()
+	}
+	return nil, fmt.Errorf("storage type %s doesn't have a CF template", o.storageType)
+}
+
+func (o *initStorageOpts) wkldDDBAddonBlobs() ([]addonBlob, error) {
+	props, err := o.ddbProps()
 	if err != nil {
 		return nil, err
 	}
-	blobs := []addonBlob{
+	return []addonBlob{
 		{
-			name:        o.storageName,
+			path:        o.ws.WorkloadAddonFilePath(o.workloadName, fmt.Sprintf("%s.yml", o.storageName)),
 			description: "template",
-			blob:        templateBlob,
+			blob:        addon.WorkloadDDBTemplate(props),
 		},
-	}
-	paramsBlob, err := o.newAddonParams()
-	if err != nil {
-		if errors.Is(err, errUnavailableAddonParams) { // The addon does not need any parameters.
-			return blobs, nil
-		}
-		return nil, err
-	}
-	return append(blobs, addonBlob{
-		name:        "addons.parameters",
-		description: "parameters",
-		blob:        paramsBlob,
-	}), nil
+	}, nil
 }
 
-func (o *initStorageOpts) newAddonTemplate() (encoding.BinaryMarshaler, error) {
-	var templateBlob encoding.BinaryMarshaler
-	err := fmt.Errorf("storage type %s doesn't have a CF template", o.storageType)
-	switch o.storageType {
-	case dynamoDBStorageType:
-		templateBlob, err = o.newDDBTemplate()
-	case s3StorageType:
-		templateBlob, err = o.newS3Template()
-	case rdsStorageType:
-		templateBlob, err = o.newRDSTemplate()
-	}
+func (o *initStorageOpts) envDDBAddonBlobs() ([]addonBlob, error) {
+	props, err := o.ddbProps()
 	if err != nil {
 		return nil, err
 	}
-	return templateBlob, nil
-}
-
-func (o *initStorageOpts) newAddonParams() (encoding.BinaryMarshaler, error) {
-	if o.storageType != rdsStorageType {
-		return nil, errUnavailableAddonParams
-	}
-	if o.workloadType != manifest.RequestDrivenWebServiceType {
-		return nil, errUnavailableAddonParams
-	}
-	return addon.RDWSParamsForRDS(), nil
+	return []addonBlob{
+		{
+			path:        o.ws.EnvAddonFilePath(fmt.Sprintf("%s.yml", o.storageName)),
+			description: "template",
+			blob:        addon.EnvDDBTemplate(props),
+		},
+		{
+			path:        o.ws.WorkloadAddonFilePath(o.workloadName, fmt.Sprintf("%s-access-policy.yml", o.storageName)),
```

Comment:
Wouldn't it be possible to define an env addon for DDB in a Git repository that contains only environment manifests?

For example, what if my git repository is just this:
```
copilot/
  environments/
      test/manifest.yml
      prod/manifest.yml
```
Should we still let users create this template and instead provide this addon blob as a recommended action?

Reply:
reiterating the offline discussion here for visibility - in the ⬆️  case, we will:
1. not write the addon template to `copilot/<svc>/addons/ingress.yml
2. will provide a recommended actions to add the ingress

This logic is updated in the current iteration!